packages/prosemirror-resizable-view/src/prosemirror-resizable-view.ts

```diff
@@ -1,172 +1,278 @@
+          this.dom.style.height = `${startHeight + diffY}px`;
+          break;
+        case ResizableHandleType.BottomRight:
+          this.dom.style.width = `${startWidth + diffX}px`;
+          this.dom.style.height = `${startHeight + diffY}px`;
+
+          break;
+        case ResizableHandleType.BottomLeft:
+          this.dom.style.width = `${startWidth - diffX}px`;
+          this.dom.style.height = `${startHeight + diffY}px`;
+          break;
+      }
     });
 
-    outer.append(handle);
-    outer.append(inner);
+    const onMouseUp = (e: MouseEvent) => {
+      e.preventDefault();
+      handle.dataSetDragging();
+      handle.setHandleDisplay();
+      this.#element.style.pointerEvents = 'auto';
+
+      document.removeEventListener('mousemove', onMouseMove);
+      document.removeEventListener('mouseup', onMouseUp);
+      const transaction = view.state.tr.setNodeMarkup(getPos(), undefined, {
+        src: this.#node.attrs.src,
+        width: this.dom.style.width,
+        height: this.dom.style.height,
+      });
 
-    this.dom = outer;
-    this.inner = inner;
-    this.node = node;
+      this.#width = this.dom.style.width;
+      this.#height = this.dom.style.height;
+
+      view.dispatch(transaction);
+    };
+
+    document.addEventListener('mousemove', onMouseMove);
+    document.addEventListener('mouseup', onMouseUp);
   }
 
+  /**
+   * `update` will be called by Prosemirror, when the view is updating itself.
+   */
   update(node: ProsemirrorNode): boolean {
-    if (node.type !== this.node.type) {
+    if (node.type !== this.#node.type) {
       return false;
     }
 
-    if (node.attrs.width && node.attrs.width !== this.width) {
+    if (
+      this.aspectRatio === ResizableRatioType.Fixed &&
+      node.attrs.width &&
+      node.attrs.width !== this.#width
+    ) {
       return false;
     }
 
-    if (!isEqualWithoutAttrs(this.node, node, ['width'])) {
+    if (
+      this.aspectRatio === ResizableRatioType.Flexible &&
+      node.attrs.width &&
+      node.attrs.height &&
+      node.attrs.width !== this.#width &&
+      node.attrs.height !== this.#height
+    ) {
       return false;
     }
 
-    node.attrs.width = this.node = node;
-    this.width = node.attrs.width;
-
-    this.inner.style.width = node.attrs.width;
-    return true;
-  }
-}
-
-function getWidthFromNode(node: ProsemirrorNode): number {
-  const width = node.attrs.width;
+    if (!isEqualWithoutAttrs(this.#node, node, ['width', 'height'])) {
+      return false;
+    }
 
-  if (isNumber(width)) {
-    return width;
-  }
+    this.#node = node;
 
-  if (isString(width)) {
-    const w = width.match(/(\d+)px/)?.[0];
+    this.#width = node.attrs.width;
+    this.#height = node.attrs.height;
 
-    if (w) {
-      return Number.parseFloat(w);
-    }
+    return true;
   }
-
-  return 0;
 }
 
-function getSizeFromDom(element: HTMLElement | null): [number, number] {
+function getSizeFromDom(element: HTMLElement | null): DOMRect | null {
```

Comment:
We don't need `getSizeFromDom` anymore since we can just write one-line code: `element?. getBoundingClientRect ()`